src/mp4box/avc1.rs

```diff
@@ -103,6 +103,11 @@ impl<R: Read + Seek> ReadBox<&mut R> for Avc1Box {
 
         let header = BoxHeader::read(reader)?;
         let BoxHeader { name, size: s } = header;
+        if s > size {
```

Comment:
I suppose we could be more precise here (and below), with:

```Rust
if s > size.saturating_sub(reader.stream_position() - start) {
```

Do you think it's worth it to do so? Maybe not, since I think these checks are mostly to prevent crashes due to large allocations, right? In that case, maybe add comments explaining why the checks are imprecise.

This also makes me realize that pretty much every `read_box` should be doing:

```Rust
let mut reader = reader.take(size - HEADER_SIZE);
```

so that they don't read past the end of the box. But I can file a bug for that to do in a separate PR (since this one is getting large).